tests/lib/Linter.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @fileoverview Test file for Linter class
+ * @author Gyandeep Singh
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const assert = require("chai").assert,
+    Linter = require("../../lib/Linter");
+
+//------------------------------------------------------------------------------
+// Tests
+// All the core logic has been tested inside eslint.js file
+// Here we will be focusing more on the mutability portion of it
+//------------------------------------------------------------------------------
+
+/**
+ * extract the keys into an array
+ * @param {Map} mapObj - Map type object
+ * @returns {Array<*>} collection of all the keys
+ * @private
+ */
+function extractMapKeys(mapObj) {
```

Comment:
Nitpick: I think this function can be replaced with `Array.from(mapObj.keys())`.